src/Srclib/Converter.hs

```diff
@@ -64,7 +64,7 @@ toSourceUnit leaveUnfiltered ProjectResult{..} =
     renderedPath = toText (toFilePath projectResultPath)
 
     filteredGraph :: Graphing Dependency
-    filteredGraph = Graphing.shrink ff projectResultGraph
+    filteredGraph = Graphing.shrinkWithoutPromotionToDirect ff projectResultGraph
```

Comment:
This is needed because:

```
      -- 1 -> 2 -> 5 -> 6
      --      \    \
      --       \    7
      -- 3 ----> 4

Scenario:
(3) is direct and dev dep
(4) is indirect and prod dep
```

Without this fix, for case without `--include-unused-dep`, we will report incorrect direct dependency (e.g. 4).